File: Maths/Statistics/Standard_Deviation/Standard_Deviation.swift

import Foundation

func getStdDeviation(input:[Int], population: Bool = true) -> Double {
    var sq_sum = 0.0
 
    //Calculate Mean of given inout array
    var sum = 0
    for number in input {
        sum += number
    }
    let mean = Double(sum) / Double(input.count)

    //calculate the sum of squared difference between mean and each element in the input
    for number in input {
        sq_sum = sq_sum + pow(Double(number) - mean, 2)
    }

    //Divisor depends on if its population stddev or sample stddev
    let div = population ? input.count : input.count - 1
    
    return sqrt(sq_sum / Double(div))
}

let input = readLine()!.split(separator: " ").map { Int(String($0))! }

let population_stdDev = getStdDeviation(input: input)
print("Population Standard Deviation for given input series is \(population_stdDev)")

let sample_stdDev = getStdDeviation(input: input, population: false)
print("Sample Standard Deviation for given input series is \(sample_stdDev)")

File: Maths/Statistics/Standard_Deviation/Standard_Deviation.c
#include <math.h>
#include <stdio.h>

//function to calculate standard deviation
float calculateSD(float data[],int n);
int main() {
    int i,n;
    printf("Enter total No. of  elements: ");
    scanf("%d", &n);
    float data[n];
    printf("Enter %d elements: ",n);
    for (i = 0; i < n; ++i)
        scanf("%f", &data[i]);
    printf("\nStandard Deviation = %.6f", calculateSD(data,n));
    return 0;
}

float calculateSD(float data[],int n) {
    float sum = 0.0, mean, SD = 0.0;
    int i;
    for (i = 0; i < n; ++i) {
        sum += data[i];
    }
    mean = sum / n;
    for (i = 0; i < n; ++i)
        SD += pow(data[i] - mean, 2);
    return sqrt(SD / n);
}
